examples/chat_filter.rs

```diff
@@ -0,0 +1,71 @@
+use std::time::Duration;
+
+use backoff::ExponentialBackoff;
+use dcs_grpc_server::rpc::dcs::hook::{FilterChatMessageRequest, StreamChatRequest};
+use dcs_grpc_server::rpc::dcs::hook_client::HookClient;
+use futures_util::future::{select, FutureExt};
+use futures_util::stream::StreamExt;
+use log::LevelFilter;
+use tonic::{transport, Request, Status};
+
+async fn run() -> Result<(), Error> {
+    let addr = "http://127.0.0.1:50051";
+    log::debug!("Connecting to gRPC server at {}", addr);
+
+    let endpoint = transport::Endpoint::from_static(addr).keep_alive_while_idle(true);
+    let mut client = HookClient::connect(endpoint).await?;
+    let mut chat_messages = client
+        .stream_chat(Request::new(StreamChatRequest {}))
+        .await?
+        .into_inner();
+
+    while let Some(msg) = chat_messages.next().await {
+        let msg = msg?;
+        client
+            .filter_chat_message(Request::new(FilterChatMessageRequest {
+                uuid: msg.uuid,
+                message: msg.message.to_uppercase(),
+            }))
+            .await?;
```

Comment:
This best shows how the implementation works. The same approach could be used to implement a slot blocking, or banning of users.

1. The client streams all chat messages, where each chat message has a UUID
2. The server expects the client to send a `FilterChatMessageRequest ` within 16ms that tells the server how to proceed with the message